### ISSUE TITLE
[KV Feedback]: Kill queue listeners on preview deployments

### ISSUE THREAD

**Issue by oliver-oloughlin** [denoland/deploy_feedback#567]
### 🔍

- [ ] Did you search for existing issues?

### Type of feedback

Feature request

### Description

The issue I am having is that when using queues on Deploy I only want my queue listeners to be active when on a production deployment. Much like how `Deno.cron()` works. The reason being is that there is some logic with my queue handlers that I definitely only want to run in production, and never simultaneously in other environments.

### Possible solution (if applicable)

I wish there was some way to either specify for a project that queues should only be active in production, or to kill any active queue listeners on any given (or any non-production) deployment.

**Comment by igorzi** [comment]
Queue listeners should be no-op on preview deployments. @oliver-oloughlin do you have a project where this is not the case?

**Comment by oliver-oloughlin** [comment]
@igorzi I believe I have now discovered the cause of the issue, or really perceived issue. I saw logs in the preview build from queue handlers. It seems the logs were simply made from the latest, still running queue handlers. While later enqueues did not re-trigger the handlers.

So while a new production deployment was created, the jobs already running in the previous production build will continue until finished. Which makes sense.
